src/com/maru/settings/desktop/MirrorTileService.java

```diff
@@ -0,0 +1,99 @@
+package com.maru.settings.desktop;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.drawable.Icon;
+import android.mperspective.Perspective;
+import android.mperspective.PerspectiveManager;
+import android.os.Build;
+import android.service.quicksettings.Tile;
+import android.service.quicksettings.TileService;
+import android.util.Log;
+import androidx.annotation.RequiresApi;
+
+import com.maru.settings.R;
+
+@RequiresApi(api = Build.VERSION_CODES.N)
+public class MirrorTileService extends TileService {
+    private static final String TAG = "MirrorTileService";
+
+    private PerspectiveManager mPerspectiveManager;
+    private boolean mIsListening;
+
+    @Override
+    public void onTileAdded() {
+        super.onTileAdded();
+        if (mPerspectiveManager == null) {
+            mPerspectiveManager = (PerspectiveManager) getSystemService(Context.PERSPECTIVE_SERVICE);
+        }
+        mPerspectiveManager.registerPerspectiveListener(new DesktopPerspectiveListener(), null);
+    }
+
+    @Override
+    public void onStartListening() {
+        super.onStartListening();
+        mIsListening = true;
+        if (mPerspectiveManager != null) {
+            updateDesktopStateInternal(mPerspectiveManager.isDesktopRunning());
+        }
+    }
+
+    @Override
+    public void onStopListening() {
+        mIsListening = false;
+        super.onStopListening();
+    }
+
+    @Override
+    public void onTileRemoved() {
+        super.onTileRemoved();
+        if (mPerspectiveManager != null) {
+            mPerspectiveManager.unregisterPerspectiveListener();
+        }
+    }
+
+    @Override
+    public void onClick() {
```

Comment:
We will call DisplayManager.enablePhoneMirroring() here. We do not even need to check whether the display is running or not. Please refer to https://github.com/maruos/android_platform_frameworks_base/blob/maru-0.6/packages/SystemUI/src/com/android/systemui/qs/tiles/MMirrorTile.java#L64.

Reply:
Done.

Reply:
Done.